vote/src/vote_state_view.rs

```diff
@@ -0,0 +1,378 @@
+use {
+    self::{
+        field_frames::{EpochCreditsListFrame, RootSlotFrame, RootSlotView, VotesListFrame},
+        field_list_view::ListView,
+        frame_v1_14_11::VoteStateFrameV1_14_11,
+        frame_v3::VoteStateFrameV3,
+    },
+    core::fmt::Debug,
+    field_frames::{AuthorizedVotersListFrame, EpochCreditsItem},
+    solana_clock::{Epoch, Slot},
+    solana_pubkey::Pubkey,
+    solana_vote_interface::state::{BlockTimestamp, Lockout},
+    std::sync::Arc,
+};
+
+mod field_frames;
+mod field_list_view;
+mod frame_v1_14_11;
+mod frame_v3;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum VoteStateViewError {
+    ParseError,
+}
+
+pub type Result<T> = core::result::Result<T, VoteStateViewError>;
+
+enum Field {
+    NodePubkey,
+    Commission,
+    Votes,
+    RootSlot,
+    AuthorizedVoters,
+    EpochCredits,
+    LastTimestamp,
+}
+
+/// A view into a serialized VoteState.
+///
+/// This struct provides access to the VoteState data without
+/// deserializing it. This is done by parsing and caching metadata
+/// about the layout of the serialized VoteState.
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "frozen-abi", derive(AbiExample))]
+pub struct VoteStateView {
+    data: Arc<Vec<u8>>,
+    frame: VoteStateFrame,
+}
+
+impl VoteStateView {
+    pub fn try_new(data: Arc<Vec<u8>>) -> Result<Self> {
+        let frame = VoteStateFrame::try_new(data.as_ref())?;
+        Ok(Self { data, frame })
+    }
+
+    pub fn node_pubkey(&self) -> &Pubkey {
+        let offset = self.frame.offset(Field::NodePubkey);
+        unsafe { &*(self.data.as_ptr().add(offset) as *const Pubkey) }
+    }
+
+    pub fn commission(&self) -> u8 {
+        let offset = self.frame.offset(Field::Commission);
+        self.data[offset]
```

Comment:
not technically `unsafe`, but would be good to comment why this cannot be out of bounds.